tests/scalers/nsq/nsq_test.go

```diff
@@ -0,0 +1,225 @@
+		DeleteKubernetesResources(t, testNamespace, data, templates)
+	})
+
+	installNSQ(t, kc)
+
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	require.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 0, 60, 1),
+		"Replica count should start out as 0")
+
+	testActivation(t, kc, data)
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc)
+}
+
+func installNSQ(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- installing NSQ ---")
+	CreateNamespace(t, kc, nsqNamespace)
+
+	_, err := ExecuteCommand("which helm")
+	require.NoErrorf(t, err, "nsq test requires helm - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf("helm repo add nsqio %s", nsqHelmRepoURL))
+	require.NoErrorf(t, err, "error while adding nsqio helm repo - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf("helm install nsq nsqio/nsq --namespace %s --set nsqadmin.enabled=false --wait", nsqNamespace))
+	require.NoErrorf(t, err, "error while installing nsq - %s", err)
+}
+
+func uninstallNSQ(t *testing.T) {
+	t.Log("--- uninstalling NSQ ---")
+	_, err := ExecuteCommand(fmt.Sprintf("helm uninstall nsq --namespace %s", nsqNamespace))
+	require.NoErrorf(t, err, "error while uninstalling nsq - %s", err)
+	DeleteNamespace(t, nsqNamespace)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:            testNamespace,
+			NSQNamespace:             nsqNamespace,
+			DeploymentName:           deploymentName,
+			JobName:                  jobName,
+			ScaledObjectName:         scaledObjectName,
+			MinReplicaCount:          minReplicaCount,
+			MaxReplicaCount:          maxReplicaCount,
+			TopicName:                topicName,
+			ChannelName:              channelName,
+			DepthThreshold:           depthThreshold,
+			ActivationDepthThreshold: activationDepthThreshold,
+		}, []Template{
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+		}
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+
+	data.MessageCount = activationDepthThreshold
+	KubectlReplaceWithTemplate(t, data, "jobTemplate", jobTemplate)
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 60)
+
+	data.MessageCount = 1 // total message count > activationDepthThreshold
+	KubectlReplaceWithTemplate(t, data, "jobTemplate", jobTemplate)
+	require.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should reach 1 in under 1 minute")
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale out ---")
+
+	// can handle depthThreshold messages per replica - using maxReplicaCount + 1 to ensure scaling to maxReplicaCount
+	data.MessageCount = depthThreshold * (maxReplicaCount + 1)
```

Comment:
We don't need to be fine with this, to ensure that the scaling happens, we can push a bigger amount.
The consumer pulls 1 message per second, so we can publish 80 messages, which means 40 seconds approx to be consumed (which is totally okey). This also gives some time in case of pulling image is required
```suggestion
	data.MessageCount = 80
```